app.js

```diff
@@ -114,13 +114,15 @@ app.use(function(err, req, res, next) {
     if (app.get('env') === 'development') {
         // send stacktrace in development mode
         stacktrace = err.stack;
-        console.log("STACK:",err.stack);
+        if (err.status >= 500 && err.status < 600) {
```

Comment:
Looks like we're logging the stack trace if between 500 and 600, then we check the status code later, and if not set, we set it to 500 and we don't log the stack trace?

I propose explicitly handling the 404's by calling `res.send` and not calling `next`.  The catch-all handler should always print a stack trace, because it's a last resort, in a "we have no idea how we got here" kinda way.